src/AppInstallerCommonCore/Manifest/ManifestSchemaValidation.cpp

```diff
@@ -99,82 +99,68 @@ namespace AppInstaller::Manifest::YamlParser
-                break;
-            default:
-                THROW_HR(HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED));
-            }
+            resourceMap = {
+                { ManifestTypeEnum::Singleton, IDX_MANIFEST_SCHEMA_V1_1_SINGLETON },
+                { ManifestTypeEnum::Version, IDX_MANIFEST_SCHEMA_V1_1_VERSION },
+                { ManifestTypeEnum::Installer, IDX_MANIFEST_SCHEMA_V1_1_INSTALLER },
+                { ManifestTypeEnum::DefaultLocale, IDX_MANIFEST_SCHEMA_V1_1_DEFAULTLOCALE },
+                { ManifestTypeEnum::Locale, IDX_MANIFEST_SCHEMA_V1_1_LOCALE },
+            };
         }
         else if (manifestVersion >= ManifestVer{ s_ManifestVersionV1 })
         {
-            switch (manifestType)
-            {
-            case AppInstaller::Manifest::ManifestTypeEnum::Singleton:
-                schemaStr = JsonSchema::LoadResourceAsString(MAKEINTRESOURCE(IDX_MANIFEST_SCHEMA_V1_SINGLETON), MAKEINTRESOURCE(MANIFESTSCHEMA_RESOURCE_TYPE));
-                break;
-            case AppInstaller::Manifest::ManifestTypeEnum::Version:
-                schemaStr = JsonSchema::LoadResourceAsString(MAKEINTRESOURCE(IDX_MANIFEST_SCHEMA_V1_VERSION), MAKEINTRESOURCE(MANIFESTSCHEMA_RESOURCE_TYPE));
-                break;
-            case AppInstaller::Manifest::ManifestTypeEnum::Installer:
-                schemaStr = JsonSchema::LoadResourceAsString(MAKEINTRESOURCE(IDX_MANIFEST_SCHEMA_V1_INSTALLER), MAKEINTRESOURCE(MANIFESTSCHEMA_RESOURCE_TYPE));
-                break;
-            case AppInstaller::Manifest::ManifestTypeEnum::DefaultLocale:
-                schemaStr = JsonSchema::LoadResourceAsString(MAKEINTRESOURCE(IDX_MANIFEST_SCHEMA_V1_DEFAULTLOCALE), MAKEINTRESOURCE(MANIFESTSCHEMA_RESOURCE_TYPE));
-                break;
-            case AppInstaller::Manifest::ManifestTypeEnum::Locale:
-                schemaStr = JsonSchema::LoadResourceAsString(MAKEINTRESOURCE(IDX_MANIFEST_SCHEMA_V1_LOCALE), MAKEINTRESOURCE(MANIFESTSCHEMA_RESOURCE_TYPE));
-                break;
-            default:
-                THROW_HR(HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED));
-            }
+            resourceMap = {
+                { ManifestTypeEnum::Singleton, IDX_MANIFEST_SCHEMA_V1_SINGLETON },
+                { ManifestTypeEnum::Version, IDX_MANIFEST_SCHEMA_V1_VERSION },
+                { ManifestTypeEnum::Installer, IDX_MANIFEST_SCHEMA_V1_INSTALLER },
+                { ManifestTypeEnum::DefaultLocale, IDX_MANIFEST_SCHEMA_V1_DEFAULTLOCALE },
+                { ManifestTypeEnum::Locale, IDX_MANIFEST_SCHEMA_V1_LOCALE },
+            };
         }
         else
         {
-            schemaStr = JsonSchema::LoadResourceAsString(MAKEINTRESOURCE(IDX_MANIFEST_SCHEMA_PREVIEW), MAKEINTRESOURCE(MANIFESTSCHEMA_RESOURCE_TYPE));
+            idx = IDX_MANIFEST_SCHEMA_PREVIEW;
+        }
+
+        if (idx == MANIFESTSCHEMA_NO_RESOURCE)

```

Comment:
>if (idx == MANIFESTSCHEMA_NO_RESOURCE) [](http://example.com/codeflow?start=8&length=38)

If you make the above change to make preview schema info a resource map, this check is not needed any more